src/jupyterlabcontroller/storage/k8s.py

```diff
@@ -44,13 +46,19 @@
 
 __all__ = ["K8sStorageClient"]
 
+FILESERVER_LOCK = Lock()
```

Comment:
I think the storage layer is the wrong architectural layer to do this synchronization. If the goal here is to prevent racing file server startup in two different requests from the same user, the locking should be at the service layer.

Currently, you have the problem that both the `FileserverManager` and the `K8sStorageClient` are per-request, which is why I moved things into a singleton `LabStateManager`. Either that should become `StateManager` and manage the user file servers as well, or there should be a `FileserverStateManager` that plays a similar role, and then you could put locking there if needed.

Reply:
I have a FileserverStateManager now, and I can move the locking into it.

Reply:
...kinda.  Without a "parent" pointer, I can at least make the StateManager's lock available to its child processes, which should do the trick.

Reply:
We can argue about where exactly the locks belong, but I have a system now that feels a little better.